Update changelog.txt (#3337)
AzureAD/microsoft-authentication-library-for-dotnet: changelog.txt

4.44.0
==========
### New Features
**Added support in MSAL for dSTS authority** See [3198](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3198).

> **Enabled Azure.Identity to benefit from MSAL.NET token cache when used for Managed identity** See [3137](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3137).

Reviewer: ```suggestion
**Enabled Azure.Identity (Azure SDK) to benefit from MSAL.NET token cache when used for Managed Identity** See [3137](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3137).
```

> **Added PPE to known discovery endpoints** See [3265](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3265).

Reviewer: I think this should be a bug fix.

```
**Fixed a bug in instance discovery by adding pre-production environment (PPE) domains to known endpoints**. See [3265](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3265).
```

> **Refactored the performance tests and create YAML pipeline to run with CI Build** See [3297](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3297).

Reviewer: Move this to a fundamentals section to the bottom.

```
### Fundamentals
**Improved automated performance microbenchmarks to better reflect common scenarios** See [3297](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3297) and [wiki](https://aka.ms/msal-net-performance-testing).
```

### Experimental Features

> **MSAL.NET now have WAM integration via MSAL Runtime with AT POP support** See [3192](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3192).

Reviewer: ```suggestion
**MSAL.NET now has new WAM integration via MSAL Runtime with support for Proof-of-Possession access tokens** See [3192](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3192) and [wiki](http://aka.ms/msal-net-wam).
```

@gladjohn What's an official public name for MSAL runtime? (It went through a few iterations... Would be good to make sure we're consistent in naming it.)

Reviewer 2: [Per our docs](https://aka.ms/msal-net-wam) "The New MSAL WAM Preview is an abstraction layer based on MSAL C++ "

> ### Bug Fix

Reviewer: ```suggestion
### Bug Fixes
```

**Improved exception handling in case of Operation Cancelled Exception** See [3283](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3283).

> **Fixed AcquireTokenSilent to not prompt for Operating System Account** See [3294](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3294).

Reviewer: ```suggestion
**Fixed `AcquireTokenSilent` to not display a login prompt unnecessarily for operating system accounts in WAM**. See [3294](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3294).
```

**Fixed NullReferenceException in IsBrokerAvailable()** See [3261](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3261).

> **Fixed Race Condition in Region Discovery** See [3277](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3277).

Reviewer: ```suggestion
**Fixed a race condition to improve stability of region autodiscovery**. See [3277](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3277).
```

4.44.0
==========
### New Features
**Added support in MSAL for dSTS authority** See [3198](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3198).
**Enabled Azure.Identity (Azure SDK) to benefit from MSAL.NET token cache when used for Managed Identity** See [3137](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3137).

### Experimental Features

> **MSAL.NET now has new WAM preview which is an abstraction layer based on MSAL C++ with support for Proof-of-Possession access tokens**. This fixes some issues with current WAM implementation. See [3192](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3192) and [wiki](http://aka.ms/msal-net-wam).

Reviewer 2: ```suggestion
**MSAL.NET now has a new WAM preview which is an abstraction layer based on MSAL C++ with support for Proof-of-Possession access tokens**. This fixes some issues with current WAM implementation. See [3192](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3192) and [wiki](http://aka.ms/msal-net-wam).
```

### Bug Fixes

> **Improved exception handling in case of Operation Cancelled Exception** See [3283](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3283).

Reviewer 2: ```suggestion
**Improved exception handling in case of an Operation Cancelled Exception** See [3283](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3283).
```
